src/provider/utils.ts

```diff
@@ -1,22 +1,32 @@
 import type { BlockNumber } from '../types';
-import { BigNumberish } from '../utils/number';
+import { BigNumberish, toBN, toHex } from '../utils/number';
 
 /**
- * TODO
+ *
  * [Reference](https://github.com/starkware-libs/cairo-lang/blob/fc97bdd8322a7df043c87c371634b26c15ed6cee/src/starkware/starknet/services/api/feeder_gateway/feeder_gateway_client.py#L148-L153)
  *
  * @param hashValue
  * @param hashField
  */
-export function formatHash() {}
+export function formatHash(hashValue: BigNumberish): string {
+  if (typeof hashValue === 'string') return hashValue;
+  return toHex(toBN(hashValue));
+}
 
 /**
- * TODO
+ *
  * [Reference](https://github.com/starkware-libs/cairo-lang/blob/fc97bdd8322a7df043c87c371634b26c15ed6cee/src/starkware/starknet/services/api/feeder_gateway/feeder_gateway_client.py#L156-L161)
  * @param txHash
  * @param txId
  */
-export function txIdentifier() {}
+export function txIdentifier(txHash?: BigNumberish, txId?: BigNumberish): string {
+  if (!txHash) {
```

Comment:
Should we add a check for txId is `undefined` also? Or maybe it's unnecessary?

Reply:
I am not sure. If you suggest adding a check, I will. Wdyt?

Reply:
I think we should solve it as described in https://github.com/0xs34n/starknet.js/pull/123/files
this way it doesn't rely on positional arguments and adds almost no complexity